config/dynamic.c

```diff
@@ -247,6 +318,10 @@ process_single_cmd(int conn_fd, lua_State *lua_state)
 	if (ret < 0)
 		return -1;
 
+	ret = process_return_message(conn_fd, dy_conf);
```

Comment:
Not all chucks will call `update_gt_lua_states_incrementally()`, so it doesn't make sense to call `process_return_message()` in `process_single_cmd()`. A better approach is to make `l_update_gt_lua_states_incrementally()` to call `process_return_message()`.

In addition, we must have `process_cmds_from_mailbox(dy_conf);` in `dyn_cfg_proc()` as you had in the original version of this patch to clear the mailbox and log the error condition to deal with cases in which the synchronization timeout.